### ISSUE TITLE
Images, populated by javascript

### ISSUE THREAD

**Issue by npelov** [hoarder-app/hoarder#699]
### Describe the feature you'd like

Some websites (wordpress - https://www.wundertech.net/installing-jellyfin-on-proxmox/) use data-X attributes to store image urls. Then a javascript populates the image url. Hoarder fails to download these images when downloading full page archive. It would be nice to have a global setting (the easiest way) that delays the data processing after the page has been loaded to give javascript a chance to do it's job. 1 or 2 seconds (configurable) should be enough.

### Describe the benefits this would bring to existing Hoarder users

This approach is implemented in a lot of wordpress sites, preventing images from being crawled correctly. 

### Can the goal of this request already be achieved via other means?

Not that I know. Setting

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

### Additional context

try url:
https://www.wundertech.net/installing-jellyfin-on-proxmox/
website screenshot:
![image](https://github.com/user-attachments/assets/cfcd3360-74ba-4768-9bfd-8cb90b475926)

crawled:
![image](https://github.com/user-attachments/assets/01d07989-8f0d-4664-9e9e-66b1be6020c2)

**Comment by kamtschatka** [comment]
this is not possible at the moment. We are using https://github.com/Y2Z/monolith, which does not execute javascript, it simply stores the html + its assets in a html archive.

**Comment by MohamedBassem** [comment]
@kamtschatka we actually do execute the javascript first in the browser before taking the html and passing it to monolith.

We already wait until network activity on the page is stable before taking the html dump. I wonder if this wait is not long enough.

**Comment by kamtschatka** [comment]
indeed. looks like it is just loaded, when it comes into the viisble area,, so waiting longer does not help. scrolling down slowly would help, but not sure if that is something we want to do . i am also not sure, why the javascript in the monolith archive does not behave the same anymore.

**Comment by MohamedBassem** [comment]
@kamtschatka If I recall correctly, I explicitly don't capture JS in monolith. I think scrolling down is something that we can do actually. I think it makes sense.

**Comment by MohamedBassem** [comment]
@npelov yeah, we probably won't infinitely scroll down, but have some small limit.

**Comment by Ardakilic** [comment]
Thank you for the issue. I searched for the term "dynamic" and could not find it.

I'm also having the same issue with this url: https://oxyllmods.square.site/tc4056 

If I could attach a PDF to the note it'd be okay for me, but it's not possible I guess.

I'm not sure if it'd even work, but if possible, before calling the monolith, maybe if something like this would be called after page loading, it'd help, at least for my case:

```js
const autoScroll = () => new Promise(resolve => {
    const i = setInterval(() => window.scrollY + window.innerHeight >= document.body.scrollHeight ? (clearInterval(i), resolve()) : window.scrollBy(0, window.innerHeight), 100); // 100 ms delay between each scroll
});

// Usage
await autoScroll();
console.log('Finished!');
```

**Comment by Ardakilic** [comment]
Heya, 

Just realized PDFs also have the same issue. Screenshot of a PDF only shows first visible page as well.

For example, if I save this PDF to Karakeep, only the first page is stored as screenshot:

https://yuuiko.github.io/iPodGuide/iPodGuidev2-1.pdf

Since I believe these are related, I simply wanted to give some additional feedback about the issue.
